nidx/nidx_protos/build.py

```diff
@@ -35,19 +35,25 @@ def pdm_build_initialize(context):
     well_known_path = resources.files("grpc_tools") / "_proto"
 
     # Compile protos
-    for proto in [
-        "src/nidx.proto",
+    for proto, has_grpc in [
+        ("src/nidx.proto", True),
+        ("src/nodereader.proto", True),
```

Comment:
This sadly still needs grpc for the abstract index node abstraction. I plan to get rid of this in a following PR (didn't add it here to manage size)